docs/scenarios_authoring/RRDataSourcesAndSinks.md

```diff
@@ -2,42 +2,65 @@
+&nbsp;
+### Scenario response in scenarios with split and for-each nodes
+If parallel branches end with response nodes and more than one response is generated (e.g there is no filtering after split), the only response which is returned is the one which was chronologically generated first. There is not way to determine which one will be generated first, so if all responses are important to you, you need to take additional measures. If no response is generated (e.g. after filtering), an error is returned (in the future this behavior may change).
 
-- Input schema
-- Output schema
+If you want to to return data from multiple parallel branches, use [union](./BasicNodes.md#union) node to merge branches and [collect](#collect) node to collect data records into a list. Your scenario should have only one response node.
 
-Nussknacker uses those schema to detect how the request and response should look like.
+If you use for-each node and need to collect all the results into a list of values, use [collect](#collect) node.
 
-![RR schema](img/rrProperties.png "RR properties")
+&nbsp;
+## Request - Response Specific Components
 
 ## Request-Response source
 
-Thanks to the [Input schema](#request-response-schema) Nussknacker knows what data to expect in the request, so it will
-hint field names and will perform additional validations on SpEL expressions involving data from the selected topic.
+The `request` source does not take any parameters - just drag it onto the scenario authoring canvas.
 
 ![RR source](img/rrSource.png "RR source")
 
+&nbsp;
 ## Request-Response sink
 
-The `response` sink configuration form will show a list of fields defined in [Input schema](#request-response-schema).
+The `response` sink configuration form will show a list of fields defined in [output schema](#schemas).
 
 ![RR sink](img/rrSink.png "Kafka sink")
 
-If you prefer to define the whole response as one expression containing value of the HTTP response instead of filling
-separate autogenerated fields, you can do that by switching `Raw editor` to `true`.
+If you prefer to define the whole response as one expression containing value of the response instead of filling
+individual fields, you can do that by setting `Raw editor` to `true`.
+
+&nbsp;
+### Collect
+
+![collect](img/collect.png)
+
+As described in the [Concepts section](#concepts), there are cases when during single invocation of the request - response scenario, multiple data records start to "flow" through the scenario. Just to reiterate, this situation happens in the following cases:
+- as the result of the execution of the [for-each](./BasicNodes.md#) node,
+- if [split](./BasicNodes.md#split) node is used
+
+In such cases, the `collect` node provides a convenient way of collecting all these records into a list. If data records are in parallel branches, a [union](./BasicNodes.md/#union) node should be used  to merge the branches first.
+
+Collect node takes two arguments:
```

Comment:
nodes have arguments or parameters? I am not 100% sure, but we might call them parameters elsewhere

Reply:
changed to parameters, definitely better.